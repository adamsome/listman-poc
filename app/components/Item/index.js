import React from 'react';

const Item = (props) => {
  return (
    <div>
      <p>{ props.title }</p>
    </div>
  )
}

export default Item
