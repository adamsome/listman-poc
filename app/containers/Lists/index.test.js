import React from 'react'
import { mount, shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import { Lists } from './index'
import Item from '../../components/Item'

it('renders null based on initial state (empty "list" array)', () => {
  let wrapper = shallow(<Lists lists={[]}/>)
  expect(wrapper.contains(<div>No Items!</div>)).toBe(true)
  expect(toJson(wrapper)).toMatchSnapshot()
})

it('renders <div> w/ correct children of type <Item>', () => {
  let wrapper = shallow(<Lists lists={[ { title: 'Test One' } ]}/>)
  expect(wrapper.find('.lists')).toHaveLength(1)

  const child = wrapper.childAt(0)
  const childProps = child.props()

  expect(child.type()).toEqual(Item)
  expect(childProps.title).toEqual('Test One')
  expect(toJson(wrapper)).toMatchSnapshot()
})

//it('calls componentDidMount', () => {
  //spyLifecycle(List);

  //const props = {
    //onMount: () => {},
    //isActive: false
  //}

  //mount(<List {...props} />);

  //expect(List.prototype.componentDidMount.calledOnce).to.be.true;
//})

