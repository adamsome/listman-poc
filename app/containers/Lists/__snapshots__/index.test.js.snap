exports[`test renders <div> w/ correct children of type <Item> 1`] = `
<div
  className="lists">
  <Item
    title="Test One" />
</div>
`;

exports[`test renders null based on initial state (empty "list" array) 1`] = `
<div>
  No Items!
</div>
`;
